### ISSUE TITLE
Conduct UAT for H4 datatable and associated pages

### ISSUE THREAD

**Issue by JonellaCulmer** [fecgov/fec-cms#5652]
**What we're after:**
Conduct UAT in week 1 of the sprint to allow for front-end updates in week 2. 

**ITEMS TO CHECK:**

Design: (datatable, committee profile pages, browse data page)
- [ ] Overall appearance
- [ ] UI/UX patterns utilized correctly
- [ ] Matches design mockups/functionality, as appropriate
- [ ] Properly linked to commitee profile pages
- [ ] Browse data page(s) are accurate

Content: (datatable, browse data page(s))
- [ ] Language

Front-end (datatable, committee profile pages)
- [ ] Code
- [ ] Identify any potential bugs
- [ ] Correct linkages

Back-end
- [ ] Data is appearing as expected

### Completion criteria
- [ ] Datatable has been checked by all subgroups and approved for launch

**Comment by kathycarothers** [comment]
My only suggestion would be to add the glossary term for "nonfederal" in this sentence. 
![image](https://github.com/fecgov/fec-cms/assets/32739251/755efb4c-582e-45a7-82a2-e5e97e98d248)

**Comment by JonellaCulmer** [comment]
@rfultz @cnlucas 

Comments from RAD, including potential data issues:

**Missing dates:**
I agree with Paul about the explanation for the missing dates, old and paper. However, I’d expect the regular disbursements, Independent Expenditures, and receipts tables should suffer from the same data issue. I’ll look into those tables to see what we do and I’d suggest being consistent with them.
 
**Filtering on two year period:**
I’m not sure what’s going on with that filter because there are H4 transactions itemized during the 2023-2024 election cycle:
- Here’s an example of a report with H4 in case you want to see if on the site, C00011197’s  [May Monthly](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.fec.gov%2Fdata%2Fcommittee%2FC00011197%2F%3Ftab%3Dfilings&data=05%7C01%7CJCulmer%40fec.gov%7Ccc82b3d8087a425e698908db62af5641%7Cee91fa706c9d45e0bb084a355de91010%7C0%7C0%7C638212277469572634%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=1zlKRhArZTT5yKZfNhCdp1%2ByBUYTRivIBptn9JJ6fnI%3D&reserved=0)
- Data from the api-stage using parameter report_year = 2023: [https://api-stage.open.fec.gov/v1/schedules/schedule_h4/?api_key=SawgVOqAaXW8wZxvglBabcXsDVihDdV1LP30eVSv&sort_hide_null=false&sort_nulls_last=true&data_type=processed&report_year=2023&sort=event_purpose_date&per_page=30](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapi-stage.open.fec.gov%2Fv1%2Fschedules%2Fschedule_h4%2F%3Fapi_key%3DSawgVOqAaXW8wZxvglBabcXsDVihDdV1LP30eVSv%26sort_hide_null%3Dfalse%26sort_nulls_last%3Dtrue%26data_type%3Dprocessed%26report_year%3D2023%26sort%3Devent_purpose_date%26per_page%3D30&data=05%7C01%7CJCulmer%40fec.gov%7Ccc82b3d8087a425e698908db62af5641%7Cee91fa706c9d45e0bb084a355de91010%7C0%7C0%7C638212277469572634%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=2YJG6AfnjXziP6C5g5VQjZ6N1JzpIvPGpqojMkn1be4%3D&reserved=0)
 
Here are a few observations that might help research this:
- In the prototype, selecting Report Time Period 2023-2024 adds “report_year=2024” to the url api request. On the regular disbursements table, filtering for Report Time Period 2023-2024 adds “two_year_transaction_period=2024” to the url api call. I can see that “two_year_transaction_period” isn’t piece of results data in the H4 output, but it appears as if the way we’ve constructed the filter in its absences isn’t pulling data correctly.
- When I try to adjust the url for the H4 data table to change “report_year=2024” to “report_year=2023” the page reloads and deletes the “report_year” parameter altogether.
 
Changes to the details panel:
 
**View Image Button:**
Can we add an open image button in the panel, something like we have on the Disbursements table? Looks like “pdf_url” isn’t a results field in the H4 API endpoint, but it would be useful to add so we can view the image on the report like we can on other data tables.
 
**Description:**  can we add Description under Transaction information on the panel. We think that every column presented on the datatable view should also be present on the panel.
 
**Total amount:** one of my RAD colleagues suggested changing “Combined federal and nonfederal share” to “Combined federal and nonfederal share (total amount)” so that a user can easily map the data table fields to the panel’s fields of information.
 
**Data issues:**
I checked some missing data fields on the table and there instances in which a field is blank on the data table and the API endpoint but the field was filled in by the committee. Here’s the image of the transaction below, the “description” provided by the committee is “Meeting Expenses”: [https://docquery.fec.gov/cgi-bin/fecimg/?202010239336472357](https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocquery.fec.gov%2Fcgi-bin%2Ffecimg%2F%3F202010239336472357&data=05%7C01%7CJCulmer%40fec.gov%7Ccc82b3d8087a425e698908db62af5641%7Cee91fa706c9d45e0bb084a355de91010%7C0%7C0%7C638212277469572634%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=9VZWPj5KrV8vtrsnLxenvg4Goy1wXAdW9hJBecrwV9s%3D&reserved=0)

**Comment by JonellaCulmer** [comment]
@bmathesonFEC 
> Description: can we add Description under Transaction information on the panel. We think that every column presented on the datatable view should also be present on the panel.

Can do!

@bmathesonFEC @cnlucas 
> View Image Button:
Can we add an open image button in the panel, something like we have on the Disbursements table? Looks like “pdf_url” isn’t a results field in the H4 API endpoint, but it would be useful to add so we can view the image on the report like we can on other data tables.

I have no problem with this. Do either of you foresee any issues with this?

@bmathesonFEC @prstoetzer @djgarr @kathycarothers 
> Total amount: one of my RAD colleagues suggested changing “Combined federal and nonfederal share” to “Combined federal and nonfederal share (total amount)” so that a user can easily map the data table fields to the panel’s fields of information.

Not a huge fan of the parens if we can avoid. How about "Combined federal and nonfederal share amout" or "Combined amount of federal and nonfederal share", and adjust the datatable column header to "Combined amount"

**Comment by djgarr** [comment]
> 

Sorry for the delayed response. My preference would be to ensure "total" made it's way in the description somehow. I'd advocate if not using parentheses, to just drop them to use "Combined federal and nonfederal share total amount” (especially given how the information is conveyed on the reporting form). That may just be me though. Not sure if @bmathesonFEC @prstoetzer @kathycarothers have other ideas

**Comment by cnlucas** [comment]
I have added the two year bug and the sorting fix to the back end [ticket](https://github.com/fecgov/openFEC/issues/5364). @rfultz mentioned pushing document images to v2, which I would also recommend.

**Comment by kathycarothers** [comment]
No parens too much on the eyes. I am fine with "Combined federal and nonfederal share total amount." @bmathesonFEC @prstoetzer @JonellaCulmer @djgarr

**Comment by cnlucas** [comment]
After doing more testing, sort_hide_null is working correctly. The reason why it functions different on the disbursement table is because we automatically add 2023-2024 as a filter. If you remove the filters it will function as the H4 functions.

**Comment by rfultz** [comment]
Is "Combined … total amount" redundant?

**Comment by djgarr** [comment]
@rfultz Nope. Because of the ways the forms and line numbers are designed, it's probably best to keep both

**Comment by PaulClark2** [comment]
Just so that I'm clear, we are considering changing the  "operating expenditure" label on the committee profile page for Form 3X filers, correct. 

Since the user will be looking at the page and see the three  items that make up the totals, can we just change the label something like "Total federal and nonfederal operating expenditures?"

![image](https://github.com/fecgov/fec-cms/assets/11817734/e2428dde-1c20-4ed9-b1b0-752c24c401db)

**Comment by JonellaCulmer** [comment]
@PaulClark2 I don't believe we discussed changing the label. The only place where we refer to totals are at the top of those sections. The other sections don't elaborate on total as I think that's the expectation for a given 2-year period. I'm also not sure we need to specify that they make up federal and nonfederal because each line below explains which is which. 

I do, however, wish we could better explain what "other" is. We tend to use that a lot. 

@rfultz I also noticed that there are dashes next to the header labels on the datatable for the new columns. If they are not sortable we should drop the dashes that suggest they are. Also, the icons to expand the details panels are gray. Can we update those to match the rest of the site? Apologies if they were gray in my mock.

**Comment by rfultz** [comment]
Today's the deadline for stakeholder requests/feedback; right now it looks like we can wrap this Monday morning
